synapse/storage/databases/main/event_push_actions.py

```diff
@@ -463,6 +465,153 @@ def add_thread_id_summary_txn(txn: LoggingTransaction) -> int:
 
         return result
 
+    async def get_unread_counts_by_room_for_user(self, user_id: str) -> Dict[str, int]:
+        """Get the notification count by room for a user. Only considers notifications,
+        no highlights or unreads, and threads are currently aggregated under their room.
+
+        This function is intentionally not cached because it is called to calculate the
+        unread badge for notifications and thus the result is expected to change.
```

Comment:
```suggestion
        This function is intentionally not cached because it is called to calculate the
        unread badge for push notifications and thus the result is expected to change.
```

Reply:
Fixup docstring